### PR TITLE
Add Query Federation Support for SAP HANA

### PR DESCRIPTION
**Pull request by DilipReddyGaddam**
Issue #, if available:

Description of changes:
Changes in this are required to support Query federation for SAP Hana connector. We have to implement custom logic of SAPHanaSqlDialect to support SQL query generation using Substrait libraries.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by burhan94 — COMMENTED** [review]
## Code Review

### 🔴 Critical Issues

1. **quoteIdentifier doesn't escape embedded quotes** (SAPHanaSqlDialect.java) — If an identifier contains `"`, the generated SQL breaks. Should escape with `identifier.replace("\"", "\"\"")`. This is also a potential SQL injection vector.

2. **Behavioral change in credential fetching** (SaphanaMetadataHandler.java) — Changed from `getDatabaseConnectionConfig().getSecret()` to `secretName` parameter. Needs verification this is intentional.

3. **Dead code** — `rewriteMaxMinExpr` is public but never called. Either integrate it or remove it.

### 🟡 Medium Issues

- Tests don't verify actual SQL output for LIMIT/OFFSET
- Mixing JUnit 4 and JUnit 5 annotations in tests
- Uses `Mockito.any()` matchers instead of specific captures
- `log.info` on every query could be noisy; consider `log.debug`

### Potential Edge Cases

- `isCatalogCasingFilterApplied`: NPE risk if `split.getProperties()` returns null
- `quoteIdentifier` with `upperCaseFilterApplied=true` returns unquoted identifier — breaks on reserved words

### 🟢 What's Good

- Custom Calcite dialect approach is correct for HANA
- CHAR_LENGTH→LENGTH rewrite is a nice touch

Please address the critical issues before merge.

File: athena-saphana/src/main/java/com/amazonaws/athena/connectors/saphana/SaphanaRecordHandler.java [hunk 1]
```diff
@@ -122,7 +122,8 @@ public CredentialsProvider createCredentialsProvider(String secretName, AwsReque
         return CredentialsProviderFactory.createCredentialProvider(
                 getDatabaseConnectionConfig().getSecret(),
```
Comment (ejeffrli) [inline athena-saphana/src/main/java/com/amazonaws/athena/connectors/saphana/SaphanaRecordHandler.java:123]: should this be using secretName instead?

File: athena-saphana/src/main/java/com/amazonaws/athena/connectors/saphana/SAPHanaSqlDialect.java [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    {
+        // SAP HANA does not support approximate distinct aggregates
+        return false;
+    }
+
+    @Override
+    public StringBuilder quoteIdentifier(StringBuilder buf, String identifier)
+    {
+        if (upperCaseFilterApplied) {
+            return buf.append(identifier);
```
Comment (ejeffrli) [inline athena-saphana/src/main/java/com/amazonaws/athena/connectors/saphana/SAPHanaSqlDialect.java:159]: should this wrap identifier in double quotes like the lowercase implementation? The tests are expecting different results for some reason

Comment (DilipReddyGaddam) [reply athena-saphana/src/main/java/com/amazonaws/athena/connectors/saphana/SAPHanaSqlDialect.java:159]: Default case in SAP Hana is uppercase and we don't need to wrap the identifier in quote. Only lowercase table names and column names needs to be quoted.

File: athena-saphana/src/main/java/com/amazonaws/athena/connectors/saphana/SaphanaRecordHandler.java (hunk 1 above)
Comment (DilipReddyGaddam) [reply athena-saphana/src/main/java/com/amazonaws/athena/connectors/saphana/SaphanaRecordHandler.java:123]: Validated the both secretName and  getDatabaseConnectionConfig().getSecret() they hold same value. this is not newly added and it works fine. I think that should be okay

**Review by ZhitongYan — APPROVED** [review]

File: athena-saphana/src/main/java/com/amazonaws/athena/connectors/saphana/SAPHanaSqlDialect.java (hunk 2 above)
Comment (ejeffrli) [reply athena-saphana/src/main/java/com/amazonaws/athena/connectors/saphana/SAPHanaSqlDialect.java:159]: what about escaping characters? The quotes for lowercase would allow complex names. Why can't both casing be quoted here?

File: athena-saphana/src/main/java/com/amazonaws/athena/connectors/saphana/SAPHanaSqlDialect.java [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+/**
+ * Calcite SQL dialect for SAP HANA (Calcite 1.40.0).
+ *
+ * Behavior:
+ *  - Unquoted identifiers are folded to UPPERCASE
+ *  - Quoted identifiers preserve case
+ *  - BOOLEAN MAX/MIN rewritten to BOOL aggregations
+ *  - OFFSET/FETCH rendered as LIMIT/OFFSET
+ */
+public class SAPHanaSqlDialect extends PostgresqlSqlDialect
```
Comment (ejeffrli) [inline athena-saphana/src/main/java/com/amazonaws/athena/connectors/saphana/SAPHanaSqlDialect.java:46]: Why are we extending PostgresqlSqlDialect and not SqlDialect? Their dialects can differ

File: athena-saphana/src/main/java/com/amazonaws/athena/connectors/saphana/SAPHanaSqlDialect.java (hunk 2 above)
Comment (DilipReddyGaddam) [reply athena-saphana/src/main/java/com/amazonaws/athena/connectors/saphana/SAPHanaSqlDialect.java:159]: Column names in the Substrait plan are always lowercase. When upperCaseFilterApplied=true, identifiers in the catalog are uppercase, so we emit them unquoted — HANA automatically folds unquoted identifiers to uppercase, matching the catalog. If we quoted them, it would preserve the lowercase casing from the plan, causing a mismatch with the actual column names in HANA.

For the lowercase path, we quote to preserve the exact casing.

Regarding escaping: the uppercase path only applies when the catalog casing filter confirms all identifiers are simple uppercase names. Any special characters in identifiers would go through the lowercase/quoted path where double-quote wrapping handles them. Reserved words are not a concern here since HANA's catalog metadata returns actual column names, not reserved keywords.

File: athena-saphana/src/main/java/com/amazonaws/athena/connectors/saphana/SAPHanaSqlDialect.java (hunk 3 above)
Comment (DilipReddyGaddam) [reply athena-saphana/src/main/java/com/amazonaws/athena/connectors/saphana/SAPHanaSqlDialect.java:46]: PostgresqlSqlDialect provides the double-quote identifier quoting and catalog casing filter logic that HANA also needs. SAPHanaSqlDialect overrides quoteIdentifier and unparseCall for HANA-specific
  behavior (uppercase unquoted identifiers, CHAR_LENGTH→LENGTH rewrite, BOOLEAN MAX/MIN rewrite). If we want to clean up the hierarchy to make the shared logic more explicit (e.g., extracting a common base), I can send a follow-up PR refactoring all connectors.

Comment (ejeffrli) [reply athena-saphana/src/main/java/com/amazonaws/athena/connectors/saphana/SAPHanaSqlDialect.java:46]: sounds good. we should extend based off shared logic

**Review by ejeffrli — APPROVED** [review]
